### PR TITLE
[Doc] Replace tensor by field for files beginning with o to s

### PR DESCRIPTION
**Pull request by Rullec**
Related issue = #1500 

This PR replaced `tensor` by `field` in the following files
- [x] odop.rst
- [x] offset.rst
- [x] overview.rst
- [x] profiler.rst
- [x] scalar_tensor.rst
- [x] snode.rst
- [x] sparse.rst
- [x] syntax.rst
- [x] syntax_sugars.rst

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: docs/scalar_tensor.rst [hunk 1]
```diff
@@ -63,34 +63,34 @@ Declaration
 
         func()
         y = ti.field(ti.f32, shape=())
-        # ERROR: cannot create tensor after kernel invocation!
+        # ERROR: cannot create field after kernel invocation!
```
Comment (archibate) [inline docs/scalar_tensor.rst:66]:
```suggestion
        # ERROR: cannot create fields after kernel invocation!
```

File: docs/scalar_tensor.rst [hunk 2]
```diff
@@ -63,34 +63,34 @@ Declaration
...
 
     .. code-block:: python
 
         x = ti.field(ti.f32, shape=())
         x[None] = 1
         y = ti.field(ti.f32, shape=())
-        # ERROR: cannot create tensor after any tensor accesses from the Python-scope!
+        # ERROR: cannot create field after any field accesses from the Python-scope!
```
Comment (archibate) [inline docs/scalar_tensor.rst:73]:
```suggestion
        # ERROR: cannot create fields after any field accesses from the Python-scope!
```

File: docs/scalar_tensor.rst [hunk 3]
```diff
@@ -63,34 +63,34 @@ Declaration
...
 
-You can access an element of the Taichi tensor by an index or indices.
+You can access an element of the Taichi field by an index or indices.
 
 .. attribute:: a[p, q, ...]
 
-    :parameter a: (Tensor) the tensor of scalars
-    :parameter p: (scalar) index of the first tensor dimension
-    :parameter q: (scalar) index of the second tensor dimension
+    :parameter a: (Field) the field of scalars
```
Comment (archibate) [inline docs/scalar_tensor.rst:83]:
```suggestion
    :parameter a: (ti.field) the field of scalars
```

File: docs/scalar_tensor.rst [hunk 4]
```diff
@@ -110,8 +110,8 @@ Meta data
 
 .. attribute:: a.shape
 
-    :parameter a: (Tensor) the tensor
-    :return: (tuple) the shape of tensor ``a``
+    :parameter a: (Field) the field
```
Comment (archibate) [inline docs/scalar_tensor.rst:113]:
```suggestion
    :parameter a: (ti.field) the field
```

File: docs/scalar_tensor.rst [hunk 5]
```diff
@@ -127,7 +127,7 @@ Meta data
 
 .. function:: a.dtype
 
-    :parameter a: (Tensor) the tensor
+    :parameter a: (Field) the field
```
Comment (archibate) [inline docs/scalar_tensor.rst:130]: r

File: docs/scalar_tensor.rst [hunk 6]
```diff
@@ -138,7 +138,7 @@ Meta data
 
 .. function:: a.parent(n = 1)
 
-    :parameter a: (Tensor) the tensor
+    :parameter a: (Field) the field
```
Comment (archibate) [inline docs/scalar_tensor.rst:141]: r

File: docs/snode.rst [hunk 7]
```diff
@@ -20,10 +20,10 @@ See :ref:`layout` for more details. ``ti.root`` is the root node of the data str
 .. function:: snode.place(x, ...)
 
     :parameter snode: (SNode) where to place
-    :parameter x: (tensor) tensor(s) to be placed
+    :parameter x: (field) field(s) to be placed
```
Comment (archibate) [inline docs/snode.rst:23]: r

File: docs/snode.rst [hunk 8]
```diff
@@ -33,12 +33,12 @@ See :ref:`layout` for more details. ``ti.root`` is the root node of the data str
         assert x.snode() == y.snode()
 
 
-.. function:: tensor.shape
+.. function:: field.shape
 
-    :parameter tensor: (Tensor)
-    :return: (tuple of integers) the shape of tensor
+    :parameter field: (Field)
```
Comment (archibate) [inline docs/snode.rst:38]: r

File: docs/snode.rst [hunk 9]
```diff
@@ -48,10 +48,10 @@ See :ref:`layout` for more details. ``ti.root`` is the root node of the data str
         x.shape # returns (3, 5, 4)
 
 
-.. function:: tensor.snode()
+.. function:: field.snode()
 
-    :parameter tensor: (Tensor)
-    :return: (SNode) the structual node where ``tensor`` is placed
+    :parameter field: (Field)
```
Comment (archibate) [inline docs/snode.rst:53]: r

File: docs/scalar_tensor.rst (hunk 5 above)
Comment (archibate) [reply docs/scalar_tensor.rst:130]:
```suggestion
    :parameter a: (ti.field) the field
```

File: docs/scalar_tensor.rst (hunk 6 above)
Comment (archibate) [reply docs/scalar_tensor.rst:141]:
```suggestion
    :parameter a: (ti.field) the field
```

File: docs/snode.rst (hunk 7 above)
Comment (archibate) [reply docs/snode.rst:23]:
```suggestion
    :parameter a: (ti.field) field(s) to be placed
```

File: docs/snode.rst (hunk 8 above)
Comment (archibate) [reply docs/snode.rst:38]:
```suggestion
    :parameter a: (ti.field)
```

File: docs/snode.rst (hunk 9 above)
Comment (archibate) [reply docs/snode.rst:53]:
```suggestion
    :parameter a: (ti.field)
```

**Comment by Rullec** [comment]
Maybe I can revise these `Field` to `ti.field` and have a reasonable formal paramter iapr? I'm simply afraid that they can make you tired.

**Pull request merged**